File: data/code_pairs/cpp/1760.cpp
#include <vector>
#include <string>

using i_long = long int;      // Adjust as per actual types used
using i_short = short int;    // Adjust as per actual types used

struct ncdr_file {
    int dummy;  // Placeholder for actual content
};

class ncdr_state {
public:
    static constexpr i_short NCDR_DEFAULT_ENT = 1024;
    static constexpr i_long NCDR_CHUNKING = 16384;

    bool init_done;
    i_long ncdr_id_stack_count;
    i_long ncdr_file_count;

    ncdr_state() : init_done(false), current_ncdr_id(-1), ncdr_id_stack_size(0),
                   ncdr_id_stack_count(0), ncdr_file_count(0), ncdr_file_total(0),
                   ncdr_file_highest(0) {}

private:
    i_long current_ncdr_id;
    std::vector<i_long> ncdr_id_stack;
    i_long ncdr_id_stack_size;
    std::string cur_nc_file; // Adjusted to 200 chars in constructor if needed

    std::vector<ncdr_file> ncdr_files;
    i_long ncdr_file_total;
    i_long ncdr_file_highest;
};

File: data/code_pairs/fortran/1760.f90
module ncdr_state
    implicit none
    integer :: current_ncdr_id = -1
    integer, dimension(:), allocatable :: ncdr_id_stack
    integer :: ncdr_id_stack_size = 0, ncdr_id_stack_count = 0
    logical :: init_done = .FALSE.
    character(len=200) :: cur_nc_file
    type ncdr_file
        integer :: dummy  ! Placeholder for actual ncdr_file content
    end type ncdr_file
    type(ncdr_file), dimension(:), allocatable :: ncdr_files
    integer :: ncdr_file_count = 0
    integer :: ncdr_file_total = 0
    integer :: ncdr_file_highest = 0
    integer, parameter :: NCDR_DEFAULT_ENT = 1024
    integer, parameter :: NCDR_CHUNKING = 16384
end module ncdr_state